store/src/java/com/zimbra/cs/imap/ImapSessionManager.java

```diff
@@ -241,7 +254,9 @@ public void run() {
                         }
                     });
                 }
+                sessions.remove(sessionsCacheKey(session));
```

Comment:
This is potentially removing a different object from the cache which happens to have been created for the same folder and may be for a newly opened select - are we happy with that?

Reply:
We were attempting to prevent the number of objects from growing without bounds.  I was attempting to find out the connection details such that each IMAP connection itself would compute keys that matched up to itself and only itself.  This would prevent what you were saying but so far I haven't been successful.  

Considering these values are only placed in this cache and then never referenced but they are used for deciding which session to page out or not.  This would just lead to early eviction possibly which would cause a reload and not a failure I wouldn't expect.

Reply:
Just looked and found another remove method `remove(key, value)` - that would probably be a safer one to use?